app/js/spatial/geo.js

```diff
@@ -95,6 +97,58 @@ export function watchLocation(callback) {
   );
 }
 
+/*
+  I think we should probably have this bit of mapping enumerateTilesAround() be executed only once 
+  and passed to different functions, because as it is now, this first part is done twice for both this function
+  and the announcer function to get all the tiles features.
+  - Kris
+*/
+// Gets the street from nearest address and returns its name and city
+export async function getCurrentRoad(locationProvider){
+  const features = await Promise.all(
+    enumerateTilesAround(locationProvider.latitude, locationProvider.longitude, locationProvider.radiusMeters)
+    .map(t => {
+      t.load();
+      return t.getFeatures();
+    })
+  )
+  .then(tileFeatures => {
+    const reduced = tileFeatures
+      // Flatten list of features across all nearby tiles
+      .reduce((acc, cur) => acc.concat(cur), [])
+      // Limit to roads
+      .filter(
+        f => f.feature_type == "highway" && 
+        f.geometry.type == "LineString" &&
+        f.feature_value in ["primary", "residential", "tertiary"]
+      );
+    return reduced;
+  });
+
+  //This pissed me off but to define a turf point you have to put longitude first T^T
+  const point = turf.point([locationProvider.longitude, locationProvider.latitude]);
+  features.forEach(road => {
+    const snap = turf.nearestPointOnLine(road, point, {units: "meters"});
+    road.distance = locationProvider.distance(
+      snap, { units: 'meters' }
+    );
+  });
+  
+  // Finds all non-duplicate roads under 5 meters
+  const nearestMap = new Map();
+  for(let i = 0; sorted[i].distance < 5; i++){
```

Comment:
Something is missing here: `sorted` is no longer defined anywhere;. Also, the condition `sorted[i].distance < 5` is going to throw an error when `sorted` is empty, i.e. when there are no roads within a 1km radius of the user.